src/ProjectTemplates/test/BlazorServerTemplateTest.cs

```diff
@@ -26,6 +26,9 @@ public BlazorServerTemplateTest(ProjectFactoryFixture projectFactory)
     [Fact]
     public Task BlazorServerTemplateWorks_NoAuth() => CreateBuildPublishAsync("blazorservernoauth");
 
+    [Fact]
+    public Task BlazorServerTemplateWorks_ProgamMainNoAuth() => CreateBuildPublishAsync("blazorservernoauth", args: new [] { "--use-program-main" });
```

Comment:
I see mostly NoAuth variants, is auth not supported with --use-program-main?

Reply:
It's supported, I just didn't expand the test suite to cover every combination as it didn't appear to already. I can add more combinations if we think it's warranted.